src/typing/typer.ml

```diff
@@ -2590,7 +2590,18 @@ and type_expr ?(mode=MGet) ctx (e,p) (with_type:WithType.t) =
 		let e = Matcher.Match.match_expr ctx e1 cases def with_type false p in
 		wrap e
 	| EReturn e ->
-		type_return ctx e with_type p
+		if not ctx.in_function then begin
+			display_error ctx "Return outside function" p;
```

Comment:
Maybe something like "Return is not allowed outside of functions" would be slightly clearer?

Reply:
Maybe, I just followed the wording below for `continue` and `break`.